source/kraken/apply_disruption.cpp

```diff
@@ -125,14 +140,38 @@ struct add_impacts_visitor : public apply_impacts_visitor {
 
     using apply_impacts_visitor::operator();
 
-    void operator()(nt::MetaVehicleJourney* mvj, const nt::Route* r = nullptr) {
+    void operator()(nt::MetaVehicleJourney* mvj, nt::Route* r = nullptr) {
         log_start_action(mvj->uri);
         if (impact->severity->effect == nt::disruption::Effect::NO_SERVICE) {
             LOG4CPLUS_TRACE(log, "canceling " << mvj->uri);
             mvj->cancel_vj(rt_level, impact->application_periods, pt_data, meta, r);
             mvj->impacted_by.push_back(impact);
+        } else if (impact->severity->effect == nt::disruption::Effect::MODIFIED_SERVICE) {
+            LOG4CPLUS_TRACE(log, "modifying " << mvj->uri);
+            auto vp = compute_vp(impact->application_periods, meta.production_date);
+            if (! r && ! mvj->get_base_vj().empty()) {
+                r = mvj->get_base_vj().at(0)->route;
+                LOG4CPLUS_TRACE(log, mvj->get_base_vj().at(0)->validity_patterns[0]->str());
+            }
+            auto nb_rt_vj = mvj->get_rt_vj().size();
+            std::string new_vj_uri = mvj->uri + ":modified:" + std::to_string(nb_rt_vj) + ":"
+                    + impact->disruption->uri;
+            auto* vj = mvj->create_discrete_vj(new_vj_uri,
+                type::RTLevel::RealTime,
+                vp,
+                r,
+                std::move(impact->aux_info.stop_times),
```

Comment:
A test on that should be great:
 - add a chaos message
 - add a delay
 - remove the chaos message

We want: the delay is still here.